### PR TITLE
ECS Agent V1.24.0 release

### PR DESCRIPTION
**Pull request by suneyz**
### Summary

ECS Agent V1.24.0 release

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 * Feature - Add support to remove containers and images that are not part of ECS tasks [#1752](https://github.com/aws/amazon-ecs-agent/pull/1752)
+* Feature - Add ARM build script [#1747](https://github.com/aws/amazon-ecs-agent/pull/1747)
+* Feature - Add configurable prometheus endpoint for agent metrics [#1743](https://github.com/aws/amazon-ecs-agent/pull/1745)
+* Feature - Add Host EC2 instance Public IPv4 address to container metadata file [#1730](https://github.com/aws/amazon-ecs-agent/pull/1730)
+* Enhancement - Docker SDK migration replacing go-dockerclient [#1743](https://github.com/aws/amazon-ecs-agent/pull/1743)
+* Enhancement - ECR image pull retry changed to 5 times [#1721](https://github.com/aws/amazon-ecs-agent/pull/1721)
+* Enhancement - Propagating Container Instance and Task Tags to Task Metadata endpoint [#1720](https://github.com/aws/amazon-ecs-agent/pull/1720)
+* Enhancement - Fetch AMI ID dynamically [#1754](https://github.com/aws/amazon-ecs-agent/pull/1754)
+* Bug - Fixed the edge case for stopping container when the StartContainer API call is timed out [#1749](https://github.com/aws/amazon-ecs-agent/pull/1749)
```
Comment (haikuoliu) [inline CHANGELOG.md]: This PR is not merged, please remove this line

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,16 @@
 # Changelog
 
-## 1.24.0-dev
+## 1.24.0
 * Feature - Configurable poll duration for container stats [@jcbowman](https://github.com/jcbowman) [#1646](https://github.com/aws/amazon-ecs-agent/pull/1646)
 * Feature - Add support to remove containers and images that are not part of ECS tasks [#1752](https://github.com/aws/amazon-ecs-agent/pull/1752)
+* Feature - Add ARM build script [#1747](https://github.com/aws/amazon-ecs-agent/pull/1747)
```
Comment (yumex93) [inline CHANGELOG.md]: I do not think we need to add this one. ARM build script is for our own process.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
+* Feature - Add configurable prometheus endpoint for agent metrics [#1743](https://github.com/aws/amazon-ecs-agent/pull/1745)
+* Feature - Add Host EC2 instance Public IPv4 address to container metadata file [#1730](https://github.com/aws/amazon-ecs-agent/pull/1730)
+* Enhancement - Docker SDK migration replacing go-dockerclient [#1743](https://github.com/aws/amazon-ecs-agent/pull/1743)
+* Enhancement - ECR image pull retry changed to 5 times [#1721](https://github.com/aws/amazon-ecs-agent/pull/1721)
+* Enhancement - Propagating Container Instance and Task Tags to Task Metadata endpoint [#1720](https://github.com/aws/amazon-ecs-agent/pull/1720)
+* Enhancement - Fetch AMI ID dynamically [#1754](https://github.com/aws/amazon-ecs-agent/pull/1754)
```
Comment (yumex93) [inline CHANGELOG.md]: This is also not needed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,8 +1,16 @@
 # Changelog
 
-## 1.24.0-dev
+## 1.24.0
 * Feature - Configurable poll duration for container stats [@jcbowman](https://github.com/jcbowman) [#1646](https://github.com/aws/amazon-ecs-agent/pull/1646)
 * Feature - Add support to remove containers and images that are not part of ECS tasks [#1752](https://github.com/aws/amazon-ecs-agent/pull/1752)
+* Feature - Add ARM build script [#1747](https://github.com/aws/amazon-ecs-agent/pull/1747)
+* Feature - Add configurable prometheus endpoint for agent metrics [#1743](https://github.com/aws/amazon-ecs-agent/pull/1745)
```
Comment (yumex93) [inline CHANGELOG.md]: Why the number in bracket is #1743 while the link is for pr 1745?

File: CHANGELOG.md (hunk 2 above)
Comment (suneyz) [reply CHANGELOG.md]: Make sense, will remove this

File: CHANGELOG.md (hunk 4 above)
Comment (suneyz) [reply CHANGELOG.md]: Good catch! will change the typo.

File: CHANGELOG.md (hunk 1 above)
Comment (suneyz) [reply CHANGELOG.md]: Will remove this

File: CHANGELOG.md (hunk 3 above)
Comment (suneyz) [reply CHANGELOG.md]: Will remove this

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,8 +1,12 @@
 # Changelog
 
-## 1.24.0-dev
+## 1.24.0
 * Feature - Configurable poll duration for container stats [@jcbowman](https://github.com/jcbowman) [#1646](https://github.com/aws/amazon-ecs-agent/pull/1646)
 * Feature - Add support to remove containers and images that are not part of ECS tasks [#1752](https://github.com/aws/amazon-ecs-agent/pull/1752)
+* Feature - Add configurable prometheus endpoint for agent metrics [#1745](https://github.com/aws/amazon-ecs-agent/pull/1745)
```
Comment (yumex93) [inline CHANGELOG.md]: I would like to change it to 'Introduce prometheus support for agent metrics'

**Pull request merged**